include/mbgl/gfx/renderable.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <mbgl/util/size.hpp>
+
+#include <memory>
+#include <cassert>
+
+namespace mbgl {
+namespace gfx {
+
+class RenderableResource {
+public:
+    RenderableResource() = default;
+    virtual ~RenderableResource() = default;
+    RenderableResource(RenderableResource&&) = delete;
+    RenderableResource(const RenderableResource&) = delete;
+    RenderableResource& operator=(RenderableResource&&) = delete;
+    RenderableResource& operator=(const RenderableResource&) = delete;
+};
+
+class Renderable {
+protected:
+    Renderable(const Size size_, std::unique_ptr<RenderableResource>&& resource_)
```

Comment:
nit: why make a local copy `const`?
`unique_ptr`  could be passed by value

Reply:
just the default way I write code to prevent accidentally modifying parameters